### PR TITLE
OV-113: Add avatar and script row to timeline

### PR DESCRIPTION
**Pull request by o-nedashkivska**
Related issue - https://github.com/BinaryStudioAcademy/bsa-2024-outreachvids/issues/113

This PR adds two separate rows to timeline: for avatars and scripts. Items in each row are placed in order. Items of different types cannot be dragged into row with another type. Updated dnd to work with new structure.

https://www.loom.com/share/52cfa60b96144f679f0fc9c1e4fcecde?sid=d887d86f-ade3-44b7-8d60-b17e1cd79bf6

Also, for now resizing will not work as expected: at the moment of resize items can overlap each other or leave an empty space. But when resize is done everything is ok. I didn't find a fast and easy way to implement this, maybe we should create a separate ticket for that.
https://www.loom.com/share/c1ac8079151b4a879cbc9f25f8866fa9?sid=8c7adb76-4d54-4c9a-90dd-6742adca911c

### CODE REVIEW

File: frontend/src/bundles/studio/components/timeline/row/avatars-row.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import { Text } from '~/bundles/common/components/components.js';
+import { RowNames } from '~/bundles/studio/enums/enums.js';
+import { type TimelineItemWithSpan } from '~/bundles/studio/types/types.js';
+
+import { Item } from '../item/item.js';
+import { Row } from './row.js';
+
+type Properties = {
+    items: Array<TimelineItemWithSpan>;
+    destinationPointerValue?: number | undefined;
```
Comment (sofiia-trokhymchuk) [inline frontend/src/bundles/studio/components/timeline/row/avatars-row.tsx:10]:
```suggestion
    destinationPointerValue?: number;
```
Please, change it everywhere

File: frontend/src/bundles/studio/components/timeline/row/row.tsx [hunk 2]
```diff
@@ -1,24 +1,49 @@
-import { type RowDefinition, useRow } from 'dnd-timeline';
+import { useRow, useTimelineContext } from 'dnd-timeline';
```
Comment (sofiia-trokhymchuk) [inline frontend/src/bundles/studio/components/timeline/row/row.tsx:1]: Move all the hooks to the common hooks

File: frontend/src/bundles/studio/components/timeline/timeline-view/timeline-view.tsx [hunk 3]
```diff
@@ -1,57 +1,50 @@
...
+    type TimelineRowsWithSpan,
+} from '~/bundles/studio/types/types.js';
 
-import { Item } from '../item/item.js';
+import { AvatarsRow } from '../row/avatars-row.js';
 import { Row } from '../row/row.js';
-import { Subrow } from '../subrow/subrow.js';
+import { ScriptsRow } from '../row/scripts-row.js';
 import { TimeAxis } from '../timeaxis/timeaxis.js';
 import { TimeCursor } from '../timecursor/timecursor.js';
```
Comment (sofiia-trokhymchuk) [inline frontend/src/bundles/studio/components/timeline/timeline-view/timeline-view.tsx:15]: Import from one place

File: frontend/src/bundles/studio/helpers/get-new-item-index.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    // eslint-disable-next-line unicorn/no-for-loop
+    for (let index = 0; index < items.length; index++) {
+        const { span, duration } = items[index] as TimelineItemWithSpan;
+
+        if (start > span.start && start < span.end) {
+            const center = span.start + duration / 2;
+
+            return (newActiveItemIndex = center > start ? index : index + 1);
+        }
+    }
```
Comment (sofiia-trokhymchuk) [inline frontend/src/bundles/studio/helpers/get-new-item-index.ts:24]: Update the loop https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/no-for-loop.md

**Review by sofiia-trokhymchuk — CHANGES_REQUESTED** [review]

File: frontend/src/bundles/studio/components/timeline/row/row.tsx (hunk 2 above)
Comment (o-nedashkivska) [reply frontend/src/bundles/studio/components/timeline/row/row.tsx]: Done

File: frontend/src/bundles/studio/components/timeline/timeline-view/timeline-view.tsx (hunk 3 above)
Comment (o-nedashkivska) [reply frontend/src/bundles/studio/components/timeline/timeline-view/timeline-view.tsx]: Done

File: frontend/src/bundles/studio/components/timeline/row/avatars-row.tsx (hunk 1 above)
Comment (o-nedashkivska) [reply frontend/src/bundles/studio/components/timeline/row/avatars-row.tsx]: I reviewed my components once again and noticed that `destinationPointer` in most of the cases is not optional, so I changed it to `destinationPointerValue: number | null`

**Review by anton-otroshchenko — DISMISSED** [review]

**Comment by o-nedashkivska** [comment]
Screenshot 2024-09-05 at 14 06 33

**Comment by o-nedashkivska** [comment]
After updated time axis:
Screenshot 2024-09-05 at 14 32 49

**Comment by o-nedashkivska** [comment]
After adding video controls:
Screenshot 2024-09-05 at 16 28 27

**Review by sofiia-trokhymchuk — DISMISSED** [review]

**Review by nikita-remeslov — DISMISSED** [review]

File: frontend/src/bundles/common/hooks/hooks.ts [hunk 5]
```diff
@@ -2,6 +2,7 @@ export { useAppDispatch } from './use-app-dispatch/use-app-dispatch.hook.js';
 export { useAppForm } from './use-app-form/use-app-form.hook.js';
 export { useAppSelector } from './use-app-selector/use-app-selector.hook.js';
 export { useFormField } from './use-form-field/use-form-field.hook.js';
+export { useItem, useRow, useTimelineContext } from 'dnd-timeline';
```
Comment (nikita-remeslov) [inline frontend/src/bundles/common/hooks/hooks.ts:5]:
```suggestion
export { useItem as useTimelineItem, useRow as useTimelineRow, useTimelineContext } from 'dnd-timeline';
```

**Review by nikita-remeslov — APPROVED** [review]

**Pull request merged**